rest/model/project.go

```diff
@@ -197,29 +197,31 @@ func (c *APIWorkstationConfig) BuildFromService(h interface{}) error {
-	BatchTime            int                  `json:"batch_time"`
-	RemotePath           *string              `json:"remote_path"`
-	Identifier           *string              `json:"identifier"`
-	DisplayName          *string              `json:"display_name"`
-	DeactivatePrevious   bool                 `json:"deactivate_previous"`
-	TracksPushEvents     bool                 `json:"tracks_push_events"`
-	PRTestingEnabled     bool                 `json:"pr_testing_enabled"`
-	DefaultLogger        *string              `json:"default_logger"`
-	CommitQueue          APICommitQueueParams `json:"commit_queue"`
-	TaskSync             APITaskSyncOptions   `json:"task_sync"`
-	Tracked              bool                 `json:"tracked"`
-	PatchingDisabled     bool                 `json:"patching_disabled"`
-	RepotrackerDisabled  bool                 `json:"repotracker_disabled"`
-	Admins               []*string            `json:"admins"`
-	DeleteAdmins         []*string            `json:"delete_admins,omitempty"`
-	NotifyOnBuildFailure bool                 `json:"notify_on_failure"`
-	Tags                 []*string            `json:"tags"`
+	Owner                       *string              `json:"owner_name"`
+	Repo                        *string              `json:"repo_name"`
+	Branch                      *string              `json:"branch_name"`
+	RepoKind                    *string              `json:"repo_kind"`
+	Enabled                     bool                 `json:"enabled"`
+	Private                     bool                 `json:"private"`
+	BatchTime                   int                  `json:"batch_time"`
+	RemotePath                  *string              `json:"remote_path"`
+	Identifier                  *string              `json:"identifier"`
+	DisplayName                 *string              `json:"display_name"`
+	DeactivatePrevious          bool                 `json:"deactivate_previous"`
+	TracksPushEvents            bool                 `json:"tracks_push_events"`
+	PRTestingEnabled            bool                 `json:"pr_testing_enabled"`
+	DefaultLogger               *string              `json:"default_logger"`
+	CommitQueue                 APICommitQueueParams `json:"commit_queue"`
+	TaskSync                    APITaskSyncOptions   `json:"task_sync"`
+	Tracked                     bool                 `json:"tracked"`
+	PatchingDisabled            bool                 `json:"patching_disabled"`
+	RepotrackerDisabled         bool                 `json:"repotracker_disabled"`
+	Admins                      []*string            `json:"admins"`
+	DeleteAdmins                []*string            `json:"delete_admins,omitempty"`
+	GitTagAuthorizedUsers       []*string            `bson:"git_tag_authorized_users" json:"git_tag_authorized_users"`
+	DeleteGitTagAuthorizedUsers []*string            `bson:"delete_git_tag_authorized_users,omitempty" json:"delete_git_tag_authorized_users,omitempty"`
```

Comment:
Could we avoid having this by making sure the client sends down the entire list of users each time?

Reply:
It is pretty awkward I agree. It's more for rest routes though: I want admins and authorized users to work the same way for the user, so either I would remove DeleteAdmins and DeleteGitTagAuthorizedUsers, or add DeleteGitTagAuthorizedUsers. I'm worried about changing functionality that users currently expect though.